drivers/adc/Kconfig.max32

```diff
@@ -0,0 +1,11 @@
+# ADI SoC ADC configuration options
+
+# Copyright (c) 2023-2024 Analog Devices, Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+config ADC_MAX32
+	bool "MAX32 ADC driver"
+	default y
+	depends on DT_HAS_ADI_MAX32_ADC_ENABLED
```

Comment:
PINCTRL need to selected too, It is not enabled on board file.
select PINCTRL

Reply:
Updated, thanks.